### ISSUE TITLE
Show proper source file when using `postcss-import`(use sourcemap)

### ISSUE THREAD

**Issue by MadLittleMods** [postcss/postcss-reporter#12]
When using [`postcss-import`](https://github.com/postcss/postcss-import), the path heading that `postcss-reporter` uses is the entry file that has all the imports. I would like to instead use the sourcemap to find out where the errors originally came from as the line numbers do not match up at all and it is hard to find where you broke a rule.

I use [`stylelint`](https://github.com/stylelint/stylelint) after [`postcss-import`](https://github.com/postcss/postcss-import) so it to see all of my styles and then `postcss-reporter` at the very bottom.

```
var postcssPluginStack = [
	atImport(),
	stylelint(stylelintConfig),
	// ... other plugins
	reporter({
		clearMessages: true
	})
];
```

![](http://i.imgur.com/l4y9JrU.png)

**Comment by ben-eb** [comment]
The issue is that postcss-reporter just assumes that all messages came from one source, which with postcss-import is not that case. postcss-import correctly updates the source file for the imported nodes.

https://github.com/postcss/postcss-reporter/blob/61b255c38e517409749c6a8c27af30ff8291e194/lib/reporter.js#L22-L23

Perhaps it should be assessed on a message by message basis?

**Comment by davidtheclark** [comment]
Fixed in https://github.com/postcss/postcss-reporter/pull/14

**Comment by davidtheclark** [comment]
Released in 1.3.0.
